pkg/controller/jobframework/reconciler.go

```diff
@@ -621,7 +624,13 @@ func (r *JobReconciler) getAncestorJobManagedByKueue(ctx context.Context, jobObj
 		if QueueNameForObject(parentJob) != "" {
 			return parentJob, nil
 		}
-		jobObj = parentJob
+		currentJob = parentJob
+		linksTraversed++
+		if linksTraversed > managedOwnersChainLimit {
+			r.record.Eventf(jobObj, corev1.EventTypeWarning, ReasonJobNestingTooDeep,
```

Comment:
Additionally, could we record it to logs?

Reply:
Good point.  Done.